data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -315,14 +315,16 @@ public List<R> getResult() {
     private void readChildren(RS rs, Object instance, Object parent, MappingContext<R> ctx) {
         if (ctx.manyAssociations != null) {
             Object id = readEntityId(rs, ctx);
-            MappingContext associatedCtx = ctx.manyAssociations.get(id);
-            if (associatedCtx == null) {
-                associatedCtx = ctx.copy();
-                R entity = (R) readEntity(rs, associatedCtx, parent, id);
-                Objects.requireNonNull(id);
-                ctx.associate(associatedCtx, id, entity);
-            } else {
-                readChildren(rs, instance, parent, associatedCtx);
+            if (id != null) {
```

Comment:
Here it can happen if left join (and has more than one to many collection) that id is not loaded so we can just skip reading it and continue reading other collection potentially.